chia/full_node/full_node_api.py

```diff
@@ -1267,7 +1270,22 @@ async def send_transaction(self, request: wallet_protocol.SendTransaction, *, te
 
     @api_request
     async def request_puzzle_solution(self, request: wallet_protocol.RequestPuzzleSolution) -> Optional[Message]:
-        coin_name = request.coin_name
+        return await self._request_puzzle_solution(request)
+
+    @api_request
+    async def request_puzzle_solution_with_coin_info(
+        self, request: wallet_protocol.RequestPuzzleSolutionWithCoinInfo
+    ) -> Optional[Message]:
+        return await self._request_puzzle_solution(request)
+
+    async def _request_puzzle_solution(
+        self,
+        request: Union[wallet_protocol.RequestPuzzleSolution, wallet_protocol.RequestPuzzleSolutionWithCoinInfo],
+    ) -> Optional[Message]:
+        if isinstance(request, wallet_protocol.RequestPuzzleSolutionWithCoinInfo):
+            coin_name: bytes32 = request.coin.name()
+        else:
+            coin_name = request.coin_name
         height = request.height
         coin_record = await self.full_node.coin_store.get_coin_record(coin_name)
```

Comment:
given `coin_record` is always fetched, doesn't that provide coin info in any case? (if provided through protocol request or not)

Reply:
that is a good observation. We can probably get away without introducing a new message type

Reply:
:) good catch